igvm/hypervisor.py

```diff
@@ -55,12 +55,22 @@ def __init__(self, dataset_obj):
                 'Hypervisor "{0}" is retired.'.format(self.fqdn)
             )
 
+        self._mount_path = {}
+
+    def get_storage_pool(self):
         # Store per-VM path information
         # We cannot store these in the VM object due to migrations.
-        self._mount_path = {}
+        if hasattr(self, 'storage_pool'):
+            return self.storage_pool
```

Comment:
This smells like you want a property `storage_pool` with a getter doing what get_storage_pool() does.

If you want to keep get_storage_pool() you should set `storage_pool` to None in the constructor and check for `not None` instead of using hasattr for a cleaner interface.

Either way you should rename the class variable to _storage_pool as you don't want people to interact with it directly.